drivers/usb/udc/udc_stm32.c

```diff
@@ -0,0 +1,1156 @@
+/*
+ * Copyright (c) 2023 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file  udc_stm32.c
+ * @brief STM32 USB device controller (UDC) driver
+ */
+
+#include <soc.h>
+#include <stm32_ll_bus.h>
+#include <stm32_ll_pwr.h>
+#include <stm32_ll_rcc.h>
+#include <stm32_ll_system.h>
+#include <string.h>
+#include <zephyr/usb/usb_device.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pinctrl.h>
+#include "stm32_hsem.h"
+
+#include "udc_common.h"
+
+#include <zephyr/logging/log.h>
+#include <zephyr/irq.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(udc_stm32, CONFIG_UDC_DRIVER_LOG_LEVEL);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_otghs)
+#define DT_DRV_COMPAT st_stm32_otghs
+#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32_otgfs)
+#define DT_DRV_COMPAT st_stm32_otgfs
+#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32_usb)
+#define DT_DRV_COMPAT st_stm32_usb
+#endif
+
+struct udc_stm32_data  {
+	PCD_HandleTypeDef pcd;
+	const struct device *dev;
+	uint32_t irq;
+	uint32_t occupied_mem;
+	void (*pcd_prepare)(const struct device *dev);
+	int (*clk_enable)(void);
+	int (*clk_disable)(void);
+};
+
+struct udc_stm32_config {
+	uint32_t num_endpoints;
+	uint32_t pma_offset;
+	uint32_t dram_size;
+	uint16_t ep0_mps;
+	uint16_t ep_mps;
+};
+
+static int udc_stm32_lock(const struct device *dev)
+{
+	return udc_lock_internal(dev, K_FOREVER);
+}
+
+static int udc_stm32_unlock(const struct device *dev)
+{
+	return udc_unlock_internal(dev);
+}
+
+#define hpcd2data(hpcd) CONTAINER_OF(hpcd, struct udc_stm32_data, pcd);
+
+void HAL_PCD_ResetCallback(PCD_HandleTypeDef *hpcd)
+{
+	struct udc_stm32_data *priv = hpcd2data(hpcd);
+
+	LOG_DBG("");
```

Comment:
I do not mind, but this kind of tracing is now unwanted.
Better something like `LOG_DBG("Submit reset event");`.